tokio/tests/io_join.rs

```diff
@@ -0,0 +1,83 @@
+#![warn(rust_2018_idioms)]
+#![cfg(all(feature = "full", not(target_os = "wasi")))] // Wasi does not support panic recovery
```

Comment:
```suggestion
#![cfg(feature = "full")]
```
You're not using panics or threads anywhere.